src/main/java/org/opensearch/performanceanalyzer/reader/MetricsEmitter.java

```diff
@@ -518,11 +516,11 @@ public static String categorizeThreadName(String threadName, Dimensions dimensio
         // emitWorkloadMetrics functions.
         // Hence these are ignored in this emitter.
         if (SEARCH_PATTERN.matcher(threadName).matches()) {
-            return null;
+            return "search";
```

Comment:
Can you check if emitAggregatedOSMetrics and emitWorkloadMetrics has these metrics?

Reply:
emitAggregatedOSMetrics is dealing with Shard level operation metrics as they work with  `ShardRequestMetricsSnapshot`(https://github.com/opensearch-project/performance-analyzer-rca/blob/main/src/main/java/org/opensearch/performanceanalyzer/reader/ShardRequestMetricsSnapshot.java#L53)

They emit shard level metrics and operations pertaining to search/write/bulk.

They are not covering the search and write threadpools at coordinator node, hence there is no overlap in the thread categorization.

I have pasted the [metrics API response](https://github.com/opensearch-project/performance-analyzer-rca/issues/66) when search thread is blocked at coordinator node. It doesn't indicate any information pertaining to shard level search threads for query or fetch operations, because they are different.